src/components/Tabs/ContactTab.jsx

```diff
@@ -45,13 +45,36 @@ const ContactTab = ({
       updateContacts(newContacts);
     };
   }
+
   function updateContact(key) {
     return (value) => {
       const newContacts = [...contacts];
       newContacts[activeContact][key] = value;
+      updateContacts(newContacts)
```

Comment:
Replaced this call to the DB which was omitted at some point - changes to roles were not stored and had no visual feedback until save button was pressed